### PR TITLE
fix: enqueue wcpay customisation script on all admin pages

### LINKED ISSUE #981: WooCommerce payments customization not applied when navigating from home
### Description

When navigating to WCPay landing screen from homescreen, it's not applying the JS customizations done in https://github.com/Automattic/wc-calypso-bridge/pull/977

https://user-images.githubusercontent.com/3747241/220664736-c8b7ad1a-b815-49c8-aabe-da881e8c0d64.mp4

### To reproduce

1. In a free trial site, ensure WCPay is not set up yet
2. Go to WooCommerce > Home
3. Click on `Payments` from the navigation bar
4. Observe customizations are not applied
5. Refresh the page
6. Customizations are applied

### PR DESCRIPTION
**Pull request by rjchow**
### Changes proposed in this Pull Request:

The trial banner customisations script for the WCPay page was only being enqueued if the `$current_page['id']` is 'wc-payments'. 

This may seem correct but as the WC Payments page is loaded as a React router route, there is no page refresh when the page is navigated to from within the React application.

This PR enqueues the script on all wc-admin React pages by checking if `PageController::is_admin_or_embed_page()` is true. 

As the script does not do anything unless the URL param for `path` matches one of the specified payments page URLs, there should not be any noticeable overhead. Just a bit of extra JS to load.

Ideally this feature should be introduced by including a slotfill in either the WC Payments plugin, or a slotfill on the React router layout that passes in the page id - but that would require a release of wc-core to be usable.

Closes #981.

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Set up a free trial instance with this plugin
2. Observe that the trial banner is displayed both when you navigate to the Payments page from the nav sidebar, and when you load the page directly.
3. Observe that no other page has these customisations applied.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Not sure if this is related to the PR, but I'm noticing this error in my console from this line https://github.com/Automattic/wc-calypso-bridge/blob/52e2c418f57846af844e71cdf343d2e5dbc07d4f/assets/scripts/free-trial-wc-payments.js#L71

image

**Comment by rjchow** [comment]
> Not sure if this is related to the PR, but I'm noticing this error in my console from this line

Nice catch @ilyasfoo ! It was actually broken on trunk and I didn't notice that the copy modification wasn't taking place 😅

File: assets/scripts/free-trial-wc-payments.js [hunk 1]
```diff
@@ -68,7 +68,8 @@
 			);
 
 			if ( stepItems.length === 3 ) {
-				const p = stepItems.querySelector( 'p' );
+				const { 2: setupCompleteStepElement } = stepItems;
+				const p = setupCompleteStepElement.querySelector( 'p' );
```
Comment (chihsuan) [inline assets/scripts/free-trial-wc-payments.js:72]: Just curious can we directly access the node element?

```suggestion
				const p = stepItems[ 2 ].querySelector( 'p' );
```

**Review by chihsuan — APPROVED** [review]
Tested well. Thanks for fixing the  "payment bugs". 👍

**Review by ilyasfoo — APPROVED** [review]
Works well, LGTM! 🚢 🚢

File: assets/scripts/free-trial-wc-payments.js (hunk 1 above)
Comment (ilyasfoo) [reply assets/scripts/free-trial-wc-payments.js:72]: Could be possible with [nth-of-type](https://caniuse.com/?search=nth-of-type), but I'm not too picky at this moment

Comment (rjchow) [reply assets/scripts/free-trial-wc-payments.js:72]: that would work too! But I like to name things so that its clear what "element 2" is 😬

**Pull request merged**